src/Symfony/Component/Validator/Constraints/Enum.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * Validates that a value is one of an enum.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Enum extends Constraint
+{
+    public const NO_SUCH_CHOICE_ERROR = '53dcc1b1-a8dd-4813-baa5-b8486ff56447';
+
+    protected const ERROR_NAMES = [
+        self::NO_SUCH_CHOICE_ERROR => 'NO_SUCH_CHOICE_ERROR',
+    ];
+
+    public string $type;
+    public array $only;
+    public array $except;
+    public string $message = 'The value you selected is not a valid choice.';
+
+    /**
+     * @param string $type   the type of the enum
+     * @param array  $only   the cases that should be considered valid
```

Comment:
Isn't this setting kind-of redundant? I mean, if I enumerate all valid cases here, I could use a `Choice` constraint instead.

Reply:
Can I get an answer here?

Reply:
The operation is similar to the Choice constraint, but I would like to put all the BackedEnum management in the same place and not add functions to the enums like :
```php
public static function values(): array
{
    return array_column(self::cases(), 'value');
}
```

Reply:
> and not add functions to the enums like

I don't understand what that method has to do with the `only` setting.

Reply:
If `only` setting is not present, you will have to do `Assert\Choice(choices: [MyEnum::YES->value, ...])`.
With this setting, I wanted to only have one place to use Enum like `Assert\BackedEnumValue(type: MyEnum::class, only: [MyEnum::YES])` that's why I added it. But yes, at the end, the check will be the same.